### PR TITLE
EDSC-2195: Adds ability to download data from a project

### PR DESCRIPTION
**Pull request by abbottry**

### CODE REVIEW

File: serverless/src/submitOrder/handler.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { getJwtToken } from '../util'
+
+import { getSecretEarthdataConfig } from '../../../sharedUtils/config'
+
+// Knex database connection object
+let dbConnection = null
+
+const submitOrder = async (event) => {
+  const { body } = event
+  const { collections, environment } = body
```
Comment (macrouch) [inline serverless/src/submitOrder/handler.js]:
```
const { params } = body
const { collections, environment } = params
```

File: serverless/src/edlCallback.js [hunk 2]
```diff
@@ -31,17 +31,36 @@ export default async function edlCallback(event, context, callback) {
...
+  const jwtToken = jwt.sign({ token }, secret)
+
+  try {
+    const eventParams = {
+      username,
+      token: accessToken
+    }
+
+    // Invoke the Lambda to store the authenticated users' data in our database
+    await invokeLambda('earthdata-search-lab-storeUserData', eventParams)
```
Comment (abbottry) [inline serverless/src/edlCallback.js]: process.env?

File: serverless/src/processTag/__tests__/removeTag.test.js [hunk 3]
```diff
@@ -29,5 +29,38 @@ describe('removeTag', () => {
...
+  test('reports an error when calls to cmr fail', async () => {
+    jest.spyOn(getEarthdataConfig, 'getEarthdataConfig').mockImplementation(() => ({ cmrHost: 'http://example.com' }))
+
+    const cmrDeleteMock = jest.spyOn(request, 'delete').mockImplementation(() => {
+      throw new Error()
+    })
+
+    const consoleMock = jest.spyOn(console, 'log').mockImplementation(() => jest.fn())
+
+
```
Comment (abbottry) [inline serverless/src/processTag/__tests__/removeTag.test.js]: extra line

File: serverless/src/storeUserData/handler.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  const existingUser = await dbConnection('users').select('id').where({ urs_id: username })
+  if (existingUser.length) {
+    await dbConnection('users').update({ ...userPayload }).where({ urs_id: username })
+  } else {
+    await dbConnection('users').insert({ ...userPayload })
+  }
+
+  return {}
```
Comment (abbottry) [inline serverless/src/storeUserData/handler.js]: better response?

File: serverless/src/submitOrder/handler.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          access_method: accessMethod,
+          collection_id: id,
+          granule_params: granuleParams,
+          granule_count: granuleCount
+        })
+    })
+
+    await orderDbTransaction.commit()
+
+
```
Comment (abbottry) [inline serverless/src/submitOrder/handler.js]: extra line

File: static/src/js/actions/granules.js [hunk 6]
```diff
@@ -75,6 +89,105 @@ export const excludeGranule = data => (dispatch) => {
...
+      ...granuleParams
+    })
+      .then((response) => {
+        const { data } = response
+        const { feed } = data
+        const { entry } = feed
+
+        // Fetch the download links from the granule metadata
+        const granuleLinks = getDownloadUrls(entry)
+        console.log(granuleLinks)
```
Comment (abbottry) [inline static/src/js/actions/granules.js]: remove

File: static/src/js/reducers/__tests__/granuleDownload.test.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    const expectedState = {
+      ...initialState,
+      granuleDownloadLinks: ['http://google.jp']
+    }
+
+    expect(updateGranuleDownloadParamsReducer(undefined, action)).toEqual(expectedState)
+  })
+})
+
+describe('UPDATE_GRANULE_LINKS', () => {
```
Comment (abbottry) [inline static/src/js/reducers/__tests__/granuleDownload.test.js]: merge with above

**Pull request merged**